File: packages/contracts/src/cockpit.ts
export type CockpitTone = "neutral" | "calm" | "progress" | "success" | "warning";

export interface CockpitKpiRecord {
  id: string;
  label: string;
  value: string;
  detail: string;
  status_label: string;
  tone: CockpitTone;
}

export interface CockpitAlertRecord {
  id: string;
  title: string;
  description: string;
  module_label: string;
  tone: CockpitTone;
  priority: number;
}

export interface CockpitModuleHighlightRecord {
  id: string;
  label: string;
  value: string;
}

export interface CockpitModuleCardRecord {
  id: string;
  label: string;
  headline: string;
  detail: string;
  highlights: CockpitModuleHighlightRecord[];
  status_label: string;
  tone: CockpitTone;
}

export interface CockpitSummaryRecord {
  kpis: CockpitKpiRecord[];
  alerts: CockpitAlertRecord[];
  module_cards: CockpitModuleCardRecord[];
}

File: apps/api/app/schemas/cockpit.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class CockpitKpiRead(BaseModel):
    id: str
    label: str
    value: str
    detail: str
    status_label: str
    tone: Literal["neutral", "calm", "progress", "success", "warning"]


class CockpitAlertRead(BaseModel):
    id: str
    title: str
    description: str
    module_label: str
    tone: Literal["neutral", "calm", "progress", "success", "warning"]
    priority: int


class CockpitModuleHighlightRead(BaseModel):
    id: str
    label: str
    value: str


class CockpitModuleCardRead(BaseModel):
    id: str
    label: str
    headline: str
    detail: str
    highlights: list[CockpitModuleHighlightRead]
    status_label: str
    tone: Literal["neutral", "calm", "progress", "success", "warning"]


class CockpitSummaryRead(BaseModel):
    kpis: list[CockpitKpiRead]
    alerts: list[CockpitAlertRead]
    module_cards: list[CockpitModuleCardRead]
